### DISCUSSION TITLE
Use the file size instead of the hash of the compiler executeable on mac

### DISCUSSION THREAD

**Discussion by autoantwort** [microsoft/vcpkg#17429] [Ideas]
Currently is hashes over 350MB of file content every time you do a `vcpkg install`. In the noop case it needs 2,5 seconds on my mac. If I simply remove the hashes (yes this is no solution, only to measure time), the noop case only needs 0,5 seconds. Maybe simply use the file size in bytes of the compiler to determined if the compiler executable has changed (I would be very surprised if they manage to bring out a new version that is exactly 176'961'712 bytes in size)

**Comment by ras0219-msft** [comment]
Just to confirm: you're saying it takes 2 seconds to hash $CC?

**Reply by autoantwort** [reply]
Yes, 2 seconds for $CC and $CXX. The executable files are each 170 MB in size.

**Reply by pfaco** [reply]
I have cases in my team where the hash takes 10s to be computed. With two hashes (windows-x86-static and windows-x64) for each run. It hurts a lot.

**Comment by BurningEnlightenment** [comment]
Alternatively one could use a faster cryptographic hash algorithm (like [BLAKE3](https://github.com/BLAKE3-team/BLAKE3#readme)) for computing cache keys.
